### DISCUSSION TITLE
What's a good way to cache an API token for reuse?

### DISCUSSION THREAD

**Discussion by danechitoaie** [vercel/vercel#4156] [General] [answered]
Using Vercel + Next.js with API routes where I need to pull data from an API that requires auth which returns a token that should be used until it expires, and then refresh it. Is there a way to cache this token so I don't need to also make the auth call on every request? This token needs to remain server side, to be used only by API routes so I can't cache it client side on the users cookie/local-storage/etc.

Anyone faced the same issue?

**Comment by cycle4passion** [comment]
Not an answer, but I can say I figured out you do not have ability to write to a file. I tried using [node-persist](https://www.npmjs.com/package/node-persist). Maybe look for a similar package that keeps it in memory.

**Comment by styfle** [comment, marked as answer]
You could use a globally scoped variable in the API function like the following:

```js
let token = ''

module.exports = async (req, res) => {
  if (!token) {
    token = await generateToken()
  }
  const result = await fetchData(token)
  res.send(result)
}
```

If you need to persist the token longer, then I would recommend Redis which can assign a TTL.

**Reply by danechitoaie** [reply]
Yeah, that's a good idea. I was thinking about redis, but might be an overkill and something like this might do the trick.

**Reply by roelvan** [reply]
Google secrets might be a solution too? They have a key rotation feature. But seems too complex for your needs :D

**Reply by ghost** [reply]
You can also try [lambda.store](https://lambda.store) which is serverless redis, and free tier can be enough for your usecase.
